ext/gvl_tracing_native_extension/gvl_tracing.c

```diff
@@ -233,20 +313,21 @@ static void on_thread_event(rb_event_flag_t event_id, UNUSED_ARG const rb_intern
     case RUBY_INTERNAL_THREAD_EVENT_STARTED:   event_name = "started";   break;
     case RUBY_INTERNAL_THREAD_EVENT_EXITED:    event_name = "died";      break;
   };
-  render_event(event_name);
+  render_event(state, event_name);
 }
 
 static void on_gc_event(VALUE tpval, UNUSED_ARG void *_unused1) {
   const char* event_name = "bug_unknown_event";
+  thread_local_state *state = GT_LOCAL_STATE(rb_thread_current(), false); // no alloc during GC
   switch (rb_tracearg_event_flag(rb_tracearg_from_tracepoint(tpval))) {
     case RUBY_INTERNAL_EVENT_GC_ENTER: event_name = "gc"; break;
     // TODO: is it possible the thread wasn't running? Might need to save the last state.
     case RUBY_INTERNAL_EVENT_GC_EXIT: event_name = "running"; break;
   }
-  render_event(event_name);
+  render_event(state, event_name);
 }
```

Comment:
Minor: It may be a bit awkward right now, but at some point I've been considering maybe introducing a fake "missing" thread that could be a sink for these events, so that e.g. you at least saw in the timeline something happened, even if we couldn't attribute it to a thread.